### PR TITLE
Test DHCP Option82

### PR DESCRIPTION
**Pull request by saikrishnagajula**
### Description of PR
The PR is to verify the option82 info in DHCP relayed packets from t0 switch

Summary:
Fixes # (issue)

1.Added sniffer on leaves and added a call back on sniffed packet to look for the options info in the packet. 
2.Sniffer running on a separate thread

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
To verify option82 information in DHCP relayed packets from t0 switch

#### How did you do it?
Wrote a script to sniff relay packets on leaves and checking for relay agent information in the packet.  

#### How did you verify/test it?
Ran test_dhcp_relay.py on t0 testbed setup.

#### Any platform specific information?
currently tested on t0 setup with CEOS Neighbors

### CODE REVIEW

**Comment by kellyyeh** [comment]
@saikrishnagajula Could you please fix the check failure?

**Comment by saikrishnagajula** [comment]
> @saikrishnagajula Could you please fix the check failure?

**Comment by saikrishnagajula** [comment]
Sure will check kelly

File: ansible/roles/test/files/ptftests/dhcp_relay_test.py [hunk 1]
```diff
@@ -410,6 +413,26 @@ def client_send_discover(self, dst_mac=BROADCAST_MAC, src_port=DHCP_CLIENT_PORT)
         dhcp_discover = self.create_dhcp_discover_packet(dst_mac, src_port)
         testutils.send_packet(self, self.client_port_index, dhcp_discover)
 
+    #Verify the relayed packet has option82 info or not. Sniffing for the relayed packet on leaves and 
+    #once the packet is recieved checking for the destination and looking into options and verifying 
+    #the option82 info
+
+    def pkt_callback(self, pkt):
+        if pkt.haslayer(scapy2.IP) and pkt.haslayer(scapy2.DHCP):
+            if pkt.getlayer(scapy2.IP).dst in ['192.0.0.1','192.0.0.2','192.0.0.3','192.0.0.4'] and pkt.getlayer(scapy2.DHCP) is not None:
```
Comment (kellyyeh) [inline ansible/roles/test/files/ptftests/dhcp_relay_test.py:422]: Please retrieve dhcp server addresses dynamically

Comment (saikrishnagajula) [reply ansible/roles/test/files/ptftests/dhcp_relay_test.py]: Pushed latest code Kelly. Please Check

**Comment by kellyyeh** [comment]
@saikrishnagajula Could you please fix the merging conflict?

**Comment by saikrishnagajula** [comment]
Fixed the merge conflict Kelly

**Comment by saikrishnagajula** [comment]
@kellyyeh fixed the merge conflict issue. Please review the code

**Review by kellyyeh — APPROVED** [review]

**Pull request merged**